#! /bin/sh /usr/share/dpatch/dpatch-run
## 10_missing-header.dpatch by Ming Hua <minghua-guest@users.alioth.debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
##
## DP: Add missing header to fix FTBFS with GCC 4.3.

@DPATCH@
diff -urNad scim-pinyin-0.5.0~/src/scim_phrase.cpp scim-pinyin-0.5.0/src/scim_phrase.cpp
--- scim-pinyin-0.5.0~/src/scim_phrase.cpp	2005-01-06 07:30:58.000000000 -0600
+++ scim-pinyin-0.5.0/src/scim_phrase.cpp	2007-09-21 20:31:30.000000000 -0500
@@ -18,6 +18,7 @@
 #define Uses_STL_UTILITY
 #define Uses_STL_IOMANIP
 #define Uses_C_STDIO
+#define Uses_C_STRING
 #define Uses_SCIM_UTILITY
 #define Uses_SCIM_SERVER
 #define Uses_SCIM_ICONV
diff -urNad scim-pinyin-0.5.0~/src/scim_pinyin.cpp scim-pinyin-0.5.0/src/scim_pinyin.cpp
--- scim-pinyin-0.5.0~/src/scim_pinyin.cpp	2005-01-06 07:30:58.000000000 -0600
+++ scim-pinyin-0.5.0/src/scim_pinyin.cpp	2007-09-21 20:31:30.000000000 -0500
@@ -21,6 +21,7 @@
 #define Uses_STL_UTILITY
 #define Uses_STL_IOMANIP
 #define Uses_C_STDIO
+#define Uses_C_STRING
 #define Uses_SCIM_UTILITY
 #define Uses_SCIM_SERVER
 #define Uses_SCIM_ICONV
diff -urNad scim-pinyin-0.5.0~/src/scim_pinyin_phrase.cpp scim-pinyin-0.5.0/src/scim_pinyin_phrase.cpp
--- scim-pinyin-0.5.0~/src/scim_pinyin_phrase.cpp	2005-01-06 07:30:58.000000000 -0600
+++ scim-pinyin-0.5.0/src/scim_pinyin_phrase.cpp	2007-09-21 20:31:30.000000000 -0500
@@ -19,6 +19,7 @@
 #define Uses_STL_UTILITY
 #define Uses_STL_IOMANIP
 #define Uses_C_STDIO
+#define Uses_C_STRING
 #define Uses_SCIM_UTILITY
 #define Uses_SCIM_SERVER
 #define Uses_SCIM_ICONV
diff -urNad scim-pinyin-0.5.0~/src/scim_special_table.cpp scim-pinyin-0.5.0/src/scim_special_table.cpp
--- scim-pinyin-0.5.0~/src/scim_special_table.cpp	2005-01-06 07:31:02.000000000 -0600
+++ scim-pinyin-0.5.0/src/scim_special_table.cpp	2007-09-21 20:31:30.000000000 -0500
@@ -18,7 +18,7 @@
 #define Uses_STL_MAP
 #define Uses_STL_UTILITY
 #define Uses_STL_IOMANIP
-#define Uscs_C_STRING
+#define Uses_C_STRING
 
 #include <time.h>
 #include <scim.h>
